src/main/java/org/jabref/logic/externalfiles/AutomaticFileRenamer.java

```diff
@@ -0,0 +1,99 @@
+package org.jabref.logic.externalfiles;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.jabref.logic.FilePreferences;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.model.entry.event.FieldChangedEvent;
+
+import com.google.common.eventbus.Subscribe;
+
+public class AutomaticFileRenamer {
```

Comment:
Use org.jabref.logic.cleanup.RenamePdfCleanup#cleanup instead of self-written class. If this is not possible, comment why.